rfcs/45-stageless.md

```diff
@@ -0,0 +1,564 @@
+
+        app
+        .add_system(gravity.label(Forces))
+        .add_system_chain([broad_pass, narrow_pass].label(Physics::CollisionDetection))
+        .add_system_set([compute_forces, collision_damage], Physics::CollisionHandling)
+        // This adds the new configuration constraints to the labels in an additive fashion
+        .configure_label(Physics::Forces.configure(common_physics_config).before(Physics::CollisionDetection))
+        .configure_label(Physics::CollisionDetection.configure(common_physics_config).before(Physics::CollisionHandling))
+        .configure_label(Physics::CollisionHandling.configure(common_physics_config));
+    }
+}
+```
+
+For the most part, this is a matter of code organization and style: defining the properties of your labels up-front helps keep your code clean and your configuration centralized.
+On the other hand, configuring labels later allows you to modify the configuration of labels added by external plugins, customizing them to fit your game's needs.
+
+You can apply the same label(s) to many systems at once using **system sets** with the `App::add_system_set(systems: impl SystemIterator, labels: impl SystemLabelIterator)` method.
+
+### System ordering
+
+The most important way we can configure systems is by telling the scheduler *when* they should be run.
+The ordering of systems is always defined relative to other systems: either directly or by checking the systems that belong to a label.
+
+There are two basic forms of system ordering constraints:
+
+1. **Strict ordering constraints:** `strictly_before` and `strictly_after`
+   1. A system cannot be scheduled until "strictly before" systems must have been completed this frame.
+   2. Simple and explicit.
+   3. Can cause unneccessary blocking, particularly when systems are configured at a high-level.
+2. **If-needed ordering constraints:** `.before` and `.after`
+   1. A system cannot be scheduled until any "before" systems that it is incompatible with have completed this frame.
+   2. In 99% of cases, this is the desired behavior. Unless you are using interior mutability, systems that are compatible will always be **commutative**: their ordering doesn't matter.
+
+Applying an ordering constraint to or from a label causes a ordering constraint to be created between all individual members of that label.
+If an ordering is defined relative to a non-existent system or label, it will have no effect, emitting a warning.
+This relatively gentle failure mode is important to ensure that plugins can order their systems with relatively strong assumptions that the default system labels exist, but continue to (mostly) work if those systems or labels are not present.
+
+In addition to the `.before` and `.after` methods, you can use **system chains** to create very simple linear dependencies between the succesive members of an array of systems.

```

Comment:
Nope, we're just stealing the name to do what users think system chaining should do.

Reply:
Added a note to clarify.

Reply:
It may be worth noting what the original 'system chaining' did in the first mention of it,
I was a bit confused and only found the explanation at the end of the document.